src/opera/template/relationship.py

```diff
@@ -41,7 +48,22 @@ def get_property(self, params):
         # have data type support.
         if prop not in self.properties:
             raise DataError("Template has no '{}' property".format(prop))
+
         return self.properties[prop].eval(self, prop)
 
+    def get_attribute(self, params):
+        host, attr, *rest = params
+
+        if host != "SELF":
+            raise DataError(
+                "Accessing non-local stuff is bad. Fix your service template."
```

Comment:
This is a bit condescending to the user, let's change it to

```suggestion
                "'SELF' is the only valid value for <whatever this thing is called in tosca>"
```

Reply:
I agree that this error does not show its real source. And since this "Accessing non-local stuff is bad. Fix your service template." error is present a lot in opera, I will replace all the occurrences in a separate commit.

Reply:
This is fixed in the latest commit within this PR.